packages/babel-plugin-minify-simplify/src/index.js

```diff
@@ -28,6 +28,62 @@ module.exports = ({ types: t }) => {
   const or = (a, b) => t.logicalExpression("||", a, b);
   const and = (a, b) => t.logicalExpression("&&", a, b);
 
+  const operators = new Set([
+    '+', '-', '*', '%',
+    '<<', '>>', '>>>',
+    '&', '|', '^', '/',
+    '**'
+  ]);
+
+  const updateOperators = new Set([
+    '+', '-'
+  ]);
+
+  function isEqual(arr1, arr2) {
+    return arr1.every((value, index) => {
+      return value.toString() === arr2[index].toString();
+    });
+  }
+
+  function getName(node) {
+    if (node.type === 'ThisExpression') {
+      return 'this';
+    }
+    if (node.type === 'Super') {
+      return 'super';
+    }
+    // augment identifiers so that they don't match
+    // string/number literals
+    // but still match against each other
+    return node.name
+      ? node.name + '_'
```

Comment:
I thought about object-based abstraction but wasn't sure about perf. impact of all the "unnecessary" intermediate objects. Definitely would be less hackish.

The reason I went with duck-typing instead of explicit `isIdentifier`/`isLiteral` is in case I was forgetting some other abstraction. I'm also not sure about assumption of if something is a literal it will have a value and so on.

I just checked and I see that `NullLiteral` doesn't have value. Neither does TemplateLiteral :/

`null` also throws since I call `toString` on it instead of `String(...)`. 

Let me clean this up a little.